retrieval/scrape.go

```diff
@@ -53,6 +59,24 @@ var (
 			Help: "Total number of scrapes that were skipped because the metric storage was throttled.",
 		},
 	)
+	targetStartedScrapes = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "prometheus_target_started_scrapes_total",
+			Help: "Total number of scrapes that were attempted.",
+		},
+	)
+	targetSuccessfulScrapes = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "prometheus_target_successful_scrapes_total",
+			Help: "Total number of scrapes that were successful.",
+		},
+	)
+	targetTimeoutScrapes = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "prometheus_target_timeout_scrapes_total",
```

Comment:
this is the key new metric which allows the people to see when a scrape times out, as opposed to another kind of error